src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNQueryBuilder.java

```diff
@@ -192,6 +194,20 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
+
+        MappedFieldType mappedFieldType = context.fieldMapper(this.fieldName);
+
+        if (!(mappedFieldType instanceof KNNVectorFieldMapper.KNNVectorFieldType)) {
```

Comment:
when will this case arise?

Reply:
I dont think we check before this the MappedFieldType of the field. I thought this should be called before we call 204.